File: RestAPIs/csharp/NGO_Connect_LoginAndRegister_API/Models/BeneficiaryRequest.cs
using System;
using System.Collections.Generic;

namespace NGO_Connect_LoginAndRegister_API.Models;

public partial class BeneficiaryRequest
{
    public int RequestId { get; set; }

    public int BeneficiaryId { get; set; }

    public int ItemId { get; set; }

    public decimal? AmountNeeded { get; set; }

    public string? Description { get; set; }

    public DateTime? RequestDate { get; set; }

    public string RequestStatus { get; set; } = null!;

    public virtual User Beneficiary { get; set; } = null!;

    public virtual Item Item { get; set; } = null!;
}

File: RestAPIs/springboot/ngo-connect-springboot/src/main/java/com/ngoconnect/entities/BeneficiaryRequest.java
package com.ngoconnect.entities;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
@Entity
@Table(name = "beneficiary_request")
@Getter @Setter
public class BeneficiaryRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int requestId;

    @ManyToOne
    @JoinColumn(name = "beneficiary_id")
    private User beneficiary;

    @ManyToOne
    @JoinColumn(name = "item_id")
    private Item item;
    private String proofDocument;

    private BigDecimal amountNeeded;
    private String description;
    private String requestStatus;

    private LocalDateTime requestDate;
}
